src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,71 @@
+package duke.parser;
+
+import duke.command.*;
+
+import duke.exception.DukeException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * The parser of the user command.
+ */
+public class Parser {
+
+    /**
+     * Parses the input command.
+     *
+     * @param fullCommand The full string of user's command.
+     * @return The Command object to react.
+     * @throws DukeException When input is in wrong format.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        if (fullCommand.equals("bye")) {
+            return new ExitCommand();
+        } else if (fullCommand.equals("list")) {
+            return new ListCommand();
+        } else if (fullCommand.startsWith("done")) {
+            char num = fullCommand.charAt(fullCommand.length() - 1);
+            int index = Character.getNumericValue(num);
+            return  new DoneCommand(index);
+        } else if (fullCommand.startsWith("delete")) {
+            char num = fullCommand.charAt(fullCommand.length() - 1);
+            int index = Character.getNumericValue(num);
+            return new DeleteCommand(index);
+        } else if (fullCommand.startsWith("todo")) {
+            if (fullCommand.trim().length() == 4) {
+                throw new DukeException(" ☹ OOPS!!! The description of a todo cannot be empty.");
```

Comment:
Store the `DukeException` String as a constant final since it is being used on Lines, 39, 46, 50, 56, 60, 68.

It will make your code a lot more readable.